### PR TITLE
issue #151: guaranteed analysis title & inspector notes & various fixes and refactors

### PR DESCRIPTION
**Pull request by k-allagbe**
This PR passes the requirements for
- #149 
- #151

It merges on #159 to fix bugs in both #159 and #153.

## Changes

- Rewrote tests to use models where possible for easier modifications.

- `get_guaranteed_analysis_json`:
  - Removed the "guaranteed_analysis" field. Data is now returned directly at level 1.
  - Updated affected functions and tests.
  - Returns `null` for `title` if both `en` and `fr` are empty or null.
  - Removed unnecessary `coalesce` calls.

- `get_metrics_json`:
  - Removed the "metrics" field at level 1.
  - Returns an empty list for `weight` if it is null.

- `new_label_information`:
  - Added a trigger to convert `null` values for `title_en` and `title_fr` to empty strings on insert and update: `handle_null_titles`

- `new_inspection_function`:
  - Fixed typo in `titre`.
  - Changed `inspection_comment` to store an empty string instead of `null`.

- Schema Update (`schema_0.0.13`):
  - Renamed `user_comment` to `inspection_comment` for consistency with the model.
  - Removed unused `original_dataset` field from `inspection`.

- `update_inspection`:
  - Removed the unused `upsert_label_information` function and related tests.

- `metadata/inspection`:
  - Added a `DBInspection` model for the `inspection` table for better modifiability.
  - Simplified `build_inspection_export` by using model defaults and DB getter functions.

- Circular Dependency Fix (`queries/inspection`):
  - Fixed circular dependency between `queries/inspection` and `metadata/inspection` by removing models from `queries/inspection`.
  - Updated tests to reflect the changes.
  - Added `get_inspection_dict`, which returns an inspection entry as a dict instead of a tuple, making it easier to parse into a model. Prompted issue: [https://github.com/ai-cfia/ailab-datastore/issues/169](https://github.com/ai-cfia/ailab-datastore/issues/169).

- Blob Container Creation in `fertiscan/test_datastore`:
  - Could not get the blob container creation to work in the test code.
  - As a workaround, created the container directly in the setup.

- Created schema `fertiscan_0.0.15` to avoid conflicts with ongoing work on `fertiscan_0.0.13`.

- Rebased on main and updated `fertiscan_0.0.15`.

### CODE REVIEW

**Comment by Francois-Werbrouck** [comment]
>Blob Container Creation in fertiscan/test_datastore:
Could not get the blob container creation to work in the test code.
As a workaround, created the container directly in the setup.

This kinda scare me because it used to work. Wont your work around hide some underlying problem that we want the tests to highlight?

**Comment by k-allagbe** [comment]
> This kinda scare me because it used to work and should still work. Wont your work around hide some underlying problem that we want the tests to highlight?

Good point, Let's investigate together at you earliest convenience. Maybe I just have the wrong secrets.

**Comment by Francois-Werbrouck** [comment]
>Maybe I just have the wrong secrets.

The Github Action secrets should work and haven't been change. Please revert back all your changes for the blobs modules and tests. My issues did not impact those. Then we should investigate and see if the Actions work. If the problem persists, I suspect the problem wont be hiding in the Datastore tests/ code directly since we haven't changed a thing to our architecture.

I know the Devops team has made some changes in our ressources, if the earlier working test are still not working I'll talk with them to find out if we need to change a secret or something.

**Comment by k-allagbe** [comment]
No issue in the actions.

**Review by Francois-Werbrouck — CHANGES_REQUESTED** [review]

File: datastore/db/bytebase/FertiScan/new_inspection_function.sql [hunk 1]
```diff
@@ -314,28 +331,29 @@ BEGIN
...
-        inspector_id, label_info_id, sample_id, picture_set_id, user_comment
+    INSERT INTO "fertiscan_0.0.15".inspection (
+        inspector_id, label_info_id, sample_id, picture_set_id, inspection_comment
     ) VALUES (
         user_id, -- Assuming inspector_id is handled separately
         label_info_id,
         NULL, -- NOT handled yet
         picture_set_id,  -- Assuming picture_set_id is handled separately
-		Null
+		to_jsonb(''::text)
```
[the remark this answers is not in the document]
Comment (k-allagbe) [reply datastore/db/bytebase/FertiScan/new_inspection_function.sql]: Reverted to null. No issue detected with the frontend.

File: datastore/db/bytebase/FertiScan/new_inspection/new_label_information.sql [hunk 2]
```diff
@@ -42,3 +42,29 @@ BEGIN
     RETURN label_id;
 END;
 $function$;
+
+
+CREATE OR REPLACE FUNCTION "fertiscan_0.0.15".handle_null_titles()
```
[the remark this answers is not in the document]
Comment (k-allagbe) [reply datastore/db/bytebase/FertiScan/new_inspection/new_label_information.sql]: Reverted to null. No issue detected with the frontend.

**Pull request merged**